changelog/front-commerce-3.4-2.29.mdx

```diff
@@ -0,0 +1,471 @@
+
+:::note Previous work
+
+This API was inspired by the work of
+[@balavishnuvj/remix-seo](https://www.npmjs.com/package/@balavishnuvj/remix-seo)
+and the [NASA GCN fork](https://github.com/nasa-gcn/gcn.nasa.gov). Thanks to
+them for their contributions!
+
+:::
+
+### External Login
+
+This release introduces the ability for customers to login and register with
+external providers thanks to the new `@front-commerce/external-login-providers`
+package.
+
+We revamped how this feature was implemented under the hood and crafted public
+APIs that we hope are simple to use and extend than their 2.x counterparts:
+
+- **Providers**: Providers are responsible for handling authenticating the user
+  with the external systems. We support Google and Facebook out of the box, and
+  you can add or contribute your own providers.
+- **Adapters**: Adapters allow you to link external services users to your
+  e-commerce customer account. We support Magento2 out of the box, and Magento1
+  support will come soon.
+
+Read the
+[External Login Providers](/docs/3.x/extensions/external-login-providers/)
+documentation to learn more about this feature.
+
+### Rate Limiting
+
+We implemented a `RateLimiter` core service to limit calls on pages. This
+feature is useful to prevent abuse of your website:
+
+- Rate limiting follows the best practices and HTTP standards
+- You can use the `RateLimiter` service in your GraphQL resolvers or in your
+  Remix loaders/actions to limit the number of requests a user can make in a
+  given time frame.
+- The `RateLimiter` service uses Redis to store the rate limits in a scalable
+  way.
+- You can configure the rate limits per namespace, allowing you to have
+  different rate limits for different parts of your application.
+
+Read the
+[Prevent excessive usage with rate limits](/docs/3.x/guides/prevent-excessive-usage-with-rate-limits)
+guide to learn more about this feature.
+
+### Simplified entrypoints for standard use cases
+
+When we introduced **Front-Commerce Remixed**, our goal was to seamlessly
+integrate it into a standard Remix project. This meant that projects would have
+complete ownership of the Remix entrypoints, including the server module and
+server/client entry.
+
+However, this approach had some drawbacks. It made updates more tedious for
+projects that only used the default configuration, as they had to manually apply
+changes to their entrypoints in order to benefit from the low-level improvements
+brought by Front-Commerce.
+
+In Front-Commerce 3.4, we introduce a simpler API. We have extracted our default
+logic into the `@front-commerce/remix` package, **allowing projects to use the
+default configuration with just one line of code.**<br/> This eliminates the
+need for manual entrypoint modifications and simplifies the update process.
+
+Of course, projects still have the flexibility to "eject" from the default
+configuration for more advanced use cases.
```

Comment:
I think this is only partially true, although it would be ideal to have this also for `entry.server.tsx` and `entry.client.tsx` it's currently only true for `entry.worker.ts`, it might be good to mention that here, and to note that we aim to do the same for server and client entry points.

I would even argue that we should see if they do not exist, that we automatically inject them, so to eject from them, they need to be created in the userland code. But no need to mention that here.

Reply:
I reworded as follow:
> In Front-Commerce 3.4, we introduce a simpler API for some of them. We have
extracted our default logic for the express server and the service worker into
the `@front-commerce/remix` package, **allowing projects to use the default
configuration with just one line of code.**<br/> This eliminates the need for
manual entrypoint modifications and simplifies the update process.